src/workerd/api/sockets.c++

```diff
@@ -511,4 +512,106 @@ jsg::Ref<Socket> SocketsModule::connect(
     jsg::Lock& js, AnySocketAddress address, jsg::Optional<SocketOptions> options) {
   return connectImpl(js, kj::none, kj::mv(address), kj::mv(options));
 }
+
+kj::Own<kj::AsyncIoStream> Socket::takeConnectionStream(jsg::Lock& js) {
+  writable->detach(js);
+  readable->detach(js);
+  // We should set this before closedResolver.resolve() in order to give the user
+  // the option to check if the closed promise is resolved due to upgrade or not.
+  upgraded = true;
```

Comment:
I think this is copypasta from the code inside of Socket::startTls... I don't think we mean to set this here.